### ISSUE TITLE
Story: Implement  Page Options

### ISSUE THREAD

**Issue by VictoriaSunNIH** [NCIOCPL/cgov-digital-platform#3256]
## As a part of the redesign MVP, we need to implement Page Options

> **ESTIMATE** 5

Comp: https://app.mural.co/t/publicissapient7269/m/publicissapient7269/1652804277768/6500e76eb861e2d170cef472c81ddf30d3807a44?sender=uc0870c1f8489401535da0005

> _**NOTE:**_ in `page.html.twig` the page options should only display if the page is a Node or a Media. For example it should not display if you are viewing `/user/login`, which also uses the default page template. (There are lots of other pages that do that)

> _**NOTE:**_ This ticket includes removing the page options from the print styles. There are tickets for other hidden components (banners, breadcrumbs, left nav, page content) in NCIDS, but as page options are not an NCIDS component, the print styles need to be updated here.

### Acceptance Criteria

```gherkin
Scenario: Users are on a c.gov page
  Given users are looking at a cgov page that is not home or landing
  Then page options appear at the top of the page, inline with the breadcrumbs
    And page options also appear at the bottom above the footer
    And only 2 options appear: print and email
  ```

```gherkin
Scenario: User wants to share a c.gov page on desktop or tablet
  Given User is on a c.gov page on desktop that is not home or landing
  Then the print and email icons show in the top right under the main nav and above the footer on the left
    And onclick of the email icon, the native email client appears
    And in the email client the subject is pre-populated with "Information from the National Cancer Institute Web Site"
    And in the email client the body is pre-populated with "I found this information on www.cancer.gov and I'd like to share it with you: [enter URL]

NCI's Web site, www.cancer.gov, provides accurate, up-to-date, comprehensive cancer information from the U.S. government's principal agency for cancer research. If you have questions or need additional information, we invite you to contact NCI’s LiveHelp instant messaging service at https://livehelp.cancer.gov, or call the NCI's Contact Center 1-800-4-CANCER (1-800-422-6237) (toll-free from the United States)."
```

```gherkin
Scenario: User wants to print a c.gov page on desktop or tablet
  Given User is on a c.gov page on desktop or tablet that is not home or landing
  Then the print and email icons show in the top right under the main nav and above the footer on the left
    And onclick of the print icon, the print dialogue appears
    And the user has the choice to print the page
    And the page options do not display
```

```gherkin
Scenario: User wants to share or print a c.gov page on mobile
  Given User is on a c.gov page on mobile
  Then no icons appear for sharing or printing options
```

### Acceptance Criteria for Analytics
``` gherkin
Scenario: User on page clicks Page Options link
Given a user is on any page of cancer.gov
When the user clicks on one of the Page Options icons
Then there should be an NCIDataLayer Other event 
  And the event has the name “PageOptions:LinkClick”
  And the event has the the linkname “PageOptions:LinkClick”
  And this event should have the following data:
      | key      | value                                                   |
      | linkText | Email or Print                                          |
      | location | BodyTop if top of page, or BodyBottom if bottom of page |
```

### Notes
* The default home/landing template does not show page options - including special report.
* The app module templates do not show page options
  * App modules may provide their own page options, but those are out of scope. (i.e. R4R has their own page options within React)
* The email body needs to have the current URL, so this changes per page.
* Analytics
   * https://github.com/NCIOCPL/ncids/issues/421
* ~Print stylesheet requirements are implemented in the NCIDS as tickets:~
   * ~https://github.com/NCIOCPL/ncids/issues/493~
   * ~https://github.com/NCIOCPL/ncids/issues/494~
* Do not show page options on print, there is not a component for page options

---

## Solution
This will need to be created as:
- [ ] A Drupal block to make the links, mainly because of the url for the email
- [ ] The JS that will trigger the analytics.

**Comment by olitharp-nci** [comment]
Need to discuss print requirements?

For instance, header and footer should be hidden?

ncids_trans
[Researchers Discover Potential Way to Hit Elusive Target in Pancreatic Cancer' - National Cancer Institute.pdf](https://github.com/NCIOCPL/cgov-digital-platform/files/8575627/Researchers.Discover.Potential.Way.to.Hit.Elusive.Target.in.Pancreatic.Cancer.-.National.Cancer.Institute.pdf)

<img width="612" alt="Screen Shot 2022-04-27 at 1 49 11 PM" src="https://user-images.githubusercontent.com/85901679/165598239-97c40487-8a26-4299-abd4-841a20e32ec5.png">
<img width="618" alt="Screen Shot 2022-04-27 at 1 49 34 PM" src="https://user-images.githubusercontent.com/85901679/165598316-e6dd3c4e-dde8-4a54-aaa0-ec7a724a507c.png">

**Comment by olitharp-nci** [comment]
How should we handle page options on pages that use the special report template?

- https://www.cancer.gov/news-events/nca50?cid=wl_int_en_ccblog_nca50
- https://www.cancer.gov/research/annual-plan

**Comment by meglake** [comment]
We have approval from Lynn to remove page options from special report templates.

**Comment by bennettcc** [comment]
@dev-rana-publicis We've received the spanish translations for page options:

Page option top= Opciones de la página al inicio
Page option bottom = Opciones de la página al final

**Comment by sarinapadilla** [comment]
Closing this ticket as it was fixed by PR #3506!
